django/contrib/postgres/aggregates/general.py

```diff
@@ -1,14 +1,64 @@
 from django.contrib.postgres.fields import JSONField
 from django.db.models.aggregates import Aggregate
+from django.db.models.expressions import F, OrderBy
 
 __all__ = [
     'ArrayAgg', 'BitAnd', 'BitOr', 'BoolAnd', 'BoolOr', 'JSONBAgg', 'StringAgg',
 ]
 
 
-class ArrayAgg(Aggregate):
+class OrderableAgg(Aggregate):
+    template = "%(function)s(%(expressions)s %(ordering)s)"
+
+    def __init__(self, expression, ordering=None, **extra):
+        if not isinstance(ordering, list) and ordering is not None:
+            ordering = [ordering]
+        ordering = ordering or []
+        ordering = [(OrderBy(F(o[1:]), True) if isinstance(o, str) and o[0] == '-' else o) for o in ordering]
+        super(OrderableAgg, self).__init__(
```

Comment:
Fixed in rebase b7cf7ea48cf0785a482b5ebd04558ed9b68a9eac ([ln 18](https://github.com/django/django/pull/7604/commits/b7cf7ea48cf0785a482b5ebd04558ed9b68a9eac#diff-b1369d2e165bb1dee9ae6196389e53b0R18), [ln 23](https://github.com/django/django/pull/7604/commits/b7cf7ea48cf0785a482b5ebd04558ed9b68a9eac#diff-b1369d2e165bb1dee9ae6196389e53b0R23), [ln 33](https://github.com/django/django/pull/7604/commits/b7cf7ea48cf0785a482b5ebd04558ed9b68a9eac#diff-b1369d2e165bb1dee9ae6196389e53b0R33)
)